### ISSUE TITLE
Make "shape shifted" alg highlighting more subtle

### ISSUE THREAD

**Issue by sp3ctum** [sp3ctum/squanmate#46]
It is too dark.

**Comment by bugybunny** [comment]
I’ll continue the discussion from #45 here.

I get your points and they sound like a nice addition. If you want to have shorter scrambles for example and only care about the cubeshape and parity and nothing more. I am a fan of the asterisks but not of the highlighting as written in the other issue. It feels like an abusement of text highlighting for the wrong purpose and at least I, don’t like it from the view of UI design. I would love to hear from other people, but I think no one else is going to comment on it :D 
(Lot of I here^^) I, as a user, wondered in the beginning—when I noticed this feature—if I forgot to generate a new scramble and highlighted something before in the old case. A selected text on something newly loaded distracts and sends the user wrong signals. I know that the color is different compared to normal text selection by the user but still. 


To summarize, I feel this alg highlighting in the form of text selection is a supoptimal way. Maybe a better way would be to make this part of the alg bold or use another text formatting option.

**Comment by sp3ctum** [comment]
I like your view on this! I'll rethink and try to find out what could look
the best.

Maybe bold text or another font entirely. Or a lighter color. Lots of
options are available, but graphic design is not my strong point.

BTW I've thought about adding a marker for scrambles that misalign the
middle layer too, but I haven't been able to figure out what it could look
like. Some algs I have seen do display an asterisk at the end of the alg,
but I have already used two different colored asterisks.

**Comment by sp3ctum** [comment]
> What would be the benefit from that?
Some user on Reddit said they liked to practice different algs for cases where the middle layer is misaligned. This was back when each scramble would never flip the middle layer. Now it's random by default, which I think is good.

Also I think it's important to see these kinds of things from the algorithm when it's displayed.

I also thought about adding some indicator when the scramble arrives at cubeshape, showing which parity the puzzle has. But that would probably be a huge spoiler in the trainers.

I have some good ideas for the future 🙂

**Comment by sp3ctum** [comment]
Actually I don't think it would be a spoiler after the user notices that even scrambles never leave cubeshape, and odd scrambles always do. Maybe a minor spoiler then.

**Comment by sp3ctum** [comment]
I removed the dark gray highlighting and made the shape shifted parts of scrambles blue. I guess it's slightly better.

Bold text looked horrible, and I agree highlighting is not the way to go - it's too confusing.

**Comment by bugybunny** [comment]
> Some user on Reddit said they liked to practice different algs for cases where the middle layer is misaligned. This was back when each scramble would never flip the middle layer. Now it's random by default, which I think is good.

I agree. When learning a new case I always just learn it with the shape I'm training on top (like in your document)  like 50 times before doing the mirror. So I just henerate new scrambles until my shape is on top and I don't get the mirror. If I knew when the middle layer is flipped, I would need to generate less scrambles because I can just do a z2 during inspection.

> Actually I don't think it would be a spoiler after the user notices that even scrambles never leave cubeshape, and odd scrambles always do. Maybe a minor spoiler then.

Well, this is only the case with some basic cases like Square/Square, Kite/Kite and maybe similar fists. After that, I don't think my brain would notice that during scrambling what I'm actually doing. Because after Similar Fists you can easily swap an odd number of corners and create parity.
And for Square/Square I already see if I'll get even or odd because the scramble length gives it away.

**Comment by sp3ctum** [comment]
> So I just generate new scrambles until my shape is on top and I don't get the mirror

Oh but couldn't you just set the middle layer to always be flipped in this case?

**Comment by bugybunny** [comment]
> Oh but couldn't you just set the middle layer to always be flipped in this case?

I could but I want to train also with the top layer where I could do a y2 rotation to get a better starting position so I don’t want the middle layer to be always flipped. I guess, I start doing that for the first few times I learn a new shape, good point.

**Comment by bugybunny** [comment]
I really like this in Sq1 scramblers (and already miss it in csTimer) and I think it would really help to have this for competitions. So I created an issue for TNoodle https://github.com/thewca/tnoodle/issues/357. Just letting you know :)

**Comment by sp3ctum** [comment]
Oh this is pretty cool, thanks for letting me know. I'll keep following the
other issue to see what happens.
